test/TestExtensions/API.Test/VSSolutionHelper.cs

```diff
@@ -467,7 +467,7 @@ public static async Task<EnvDTE.Project> GetProjectAsync(EnvDTE80.Solution2 solu
             return solution2
                 .Projects
                 .Cast<EnvDTE.Project>()
-                .FirstOrDefault(project => project.Name.Equals(projectName, StringComparison.OrdinalIgnoreCase));
+                .FirstOrDefault(project => { ThreadHelper.ThrowIfNotOnUIThread(); return project.Name.Equals(projectName, StringComparison.OrdinalIgnoreCase); });
```

Comment:
I think this check `ThreadHelper.ThrowIfNotOnUIThread();` is redundant here because the caller is on the main thread already unless I am missing something basic.

Reply:
False positive by the analyzer.

See linked issue: microsoft/vs-threading#577